### DISCUSSION TITLE
Restore fails executing PVAction - MissingParameter: The request must contain the parameter zone

### DISCUSSION THREAD

**Discussion by JaHowell78** [vmware-tanzu/velero#3243] [Community support Q&A]
**What steps did you take and what happened:**
Testing a restore last night of a backup made yesterday.  Backup was successful, including creating 20 snapshots of the EBS Volumes (see velero backup describe below).  However the restore failed on nine of the volumes, all with the error of:
Cluster:  error executing PVAction for persistentvolumes/pvc-09fb8590-dc17-4306-a6d7-b6f9a92cf865: rpc error: code = Unknown desc = MissingParameter: The request must contain the parameter zone

All of the EBS volumes obviously have an AWS AZ associated to them, but in examining the volumesnapshots.json stored in the s3 bucket for the backup, those 9 are for some reason missing the volumeAZ key:value pair.

I will note, that these EBS Volumes "look" different than the others in the AWS Console, in terms of the tags, like name and others, even though they were all created dynamically by kubernetes when the PVC/PV's were created.  But none of the tags have anything to do with the AZ.  The EKS Cluster is currently on Kube 1.18, but I believe when these volume were created it was on 1.15

**What did you expect to happen:**
PVaction to succeed  and the restore to be successful

**The output of the following commands will help us better understand what's going on**:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)

**Backup Describe**
Name:         jimh-20210112182635
Namespace:    velero
Labels:       velero.io/storage-location=default
Annotations:  velero.io/source-cluster-k8s-gitversion=v1.18.9-eks-d1db3c
              velero.io/source-cluster-k8s-major-version=1
              velero.io/source-cluster-k8s-minor-version=18+

Phase:  Completed

Errors:    0
Warnings:  0

Namespaces:
  Included:  jimh
  Excluded:  <none>

Resources:
  Included:        *
  Excluded:        <none>
  Cluster-scoped:  auto

Label selector:  <none>

Storage Location:  default

Velero-Native Snapshot PVs:  auto

TTL:  36h0m0s

Hooks:  <none>

Backup Format Version:  1.1.0

Started:    2021-01-12 14:27:17 -0500 EST
Completed:  2021-01-12 14:27:42 -0500 EST

Expiration:  2021-01-14 02:27:17 -0500 EST

Total items to be backed up:  238
Items backed up:              238

Velero-Native Snapshots:  20 of 20 snapshots completed successfully (specify --details for more information)

**Restore Describe**
Name:         jimh-20210112182635-20210112182842
Namespace:    velero
Labels:       <none>
Annotations:  <none>

Phase:  PartiallyFailed (run 'velero restore logs jimh-20210112182635-20210112182842' for more information)

Started:    2021-01-12 18:28:42 -0500 EST
Completed:  2021-01-12 18:29:10 -0500 EST

Warnings:
  Velero:     <none>
  Cluster:  could not restore, customresourcedefinitions.apiextensions.k8s.io "podmonitors.monitoring.coreos.com" already exists. Warning: the in-cluster version is different than the backed-up version.
            could not restore, customresourcedefinitions.apiextensions.k8s.io "servicemonitors.monitoring.coreos.com" already exists. Warning: the in-cluster version is different than the backed-up version.
            could not restore, persistentvolumes "pvc-fa19d81d-55eb-471a-80bf-4cea4390e680" already exists. Warning: the in-cluster version is different than the backed-up version.
  Namespaces: <none>

Errors:
  Velero:     <none>
  Cluster:  error executing PVAction for persistentvolumes/pvc-09fb8590-dc17-4306-a6d7-b6f9a92cf865: rpc error: code = Unknown desc = MissingParameter: The request must contain the parameter zone
  status code: 400, request id: e3ea7c0b-09a8-44b3-96ad-3899da2463b6
    error executing PVAction for persistentvolumes/pvc-1d8a3ec0-57e7-4f36-af55-aaa9fe556fd4: rpc error: code = Unknown desc = MissingParameter: The request must contain the parameter zone
  status code: 400, request id: 256a832e-b7aa-4395-817c-a27426d0512e
    error executing PVAction for persistentvolumes/pvc-3b6d9e39-8d61-42ad-b68d-f9e9b4cf4b98: rpc error: code = Unknown desc = MissingParameter: The request must contain the parameter zone
  status code: 400, request id: 4d908eb8-1712-4430-9b1b-10d09dbb2b15
    error executing PVAction for persistentvolumes/pvc-524907f2-f8b5-4ed1-bd32-7ba56ea75cb6: rpc error: code = Unknown desc = MissingParameter: The request must contain the parameter zone
  status code: 400, request id: 6f35ed8e-de81-46e4-b3f6-6e42bd5c9baa
    error executing PVAction for persistentvolumes/pvc-63ae7616-07b8-482e-9b78-b140424a2601: rpc error: code = Unknown desc = MissingParameter: The request must contain the parameter zone
  status code: 400, request id: 00891efb-3289-4224-8fd8-db9a3eb50ed6
    error executing PVAction for persistentvolumes/pvc-6593de51-a66a-445d-a01c-5792aeff4f7e: rpc error: code = Unknown desc = MissingParameter: The request must contain the parameter zone
  status code: 400, request id: cfbed389-d192-4bc2-8157-c6d86edb51db
    error executing PVAction for persistentvolumes/pvc-73763eac-c5d6-4e97-87cf-e04c2d915189: rpc error: code = Unknown desc = MissingParameter: The request must contain the parameter zone
  status code: 400, request id: 2f86128b-d421-4534-8285-ef825871ce1c
    error executing PVAction for persistentvolumes/pvc-adc8fc17-6644-4b1a-b055-14009e05a179: rpc error: code = Unknown desc = MissingParameter: The request must contain the parameter zone
  status code: 400, request id: 70f10815-2e11-42ea-948e-474303a6e62d
    error executing PVAction for persistentvolumes/pvc-f65272d1-b6c0-4579-b999-9f5622e3ede3: rpc error: code = Unknown desc = MissingParameter: The request must contain the parameter zone
  status code: 400, request id: 27ab1581-799f-406f-9b92-9120e65bd212
  Namespaces:
    jimh2:  error restoring ingresses.extensions/jimh2/jimh-cloud-gateway: admission webhook "validate.nginx.ingress.kubernetes.io" denied the request: host "jimh-cloud-gateway.dev.ontariosystems.cloud" and path "/" is already defined in ingress jimh/jimh-cloud-gateway
            error restoring ingresses.extensions/jimh2/jimh-cloud-portal: admission webhook "validate.nginx.ingress.kubernetes.io" denied the request: host "jimh-ui.dev.ontariosystems.cloud" and path "/" is already defined in ingress jimh/jimh-cloud-portal
            error restoring ingresses.extensions/jimh2/jimh-naccel-gateway: admission webhook "validate.nginx.ingress.kubernetes.io" denied the request: host "jimh-naccel-gateway.dev.ontariosystems.cloud" and path "/" is already defined in ingress jimh/jimh-naccel-gateway
            error restoring ingresses.extensions/jimh2/jimh-naccel: admission webhook "validate.nginx.ingress.kubernetes.io" denied the request: host "jimh-naccel-background.dev.ontariosystems.cloud" and path "/" is already defined in ingress jimh/jimh-naccel
            error restoring ingresses.networking.k8s.io/jimh2/jimh-cloud-gateway: admission webhook "validate.nginx.ingress.kubernetes.io" denied the request: host "jimh-cloud-gateway.dev.ontariosystems.cloud" and path "/" is already defined in ingress jimh/jimh-cloud-gateway
            error restoring ingresses.networking.k8s.io/jimh2/jimh-cloud-portal: admission webhook "validate.nginx.ingress.kubernetes.io" denied the request: host "jimh-ui.dev.ontariosystems.cloud" and path "/" is already defined in ingress jimh/jimh-cloud-portal
            error restoring ingresses.networking.k8s.io/jimh2/jimh-naccel-gateway: admission webhook "validate.nginx.ingress.kubernetes.io" denied the request: host "jimh-naccel-gateway.dev.ontariosystems.cloud" and path "/" is already defined in ingress jimh/jimh-naccel-gateway
            error restoring ingresses.networking.k8s.io/jimh2/jimh-naccel: admission webhook "validate.nginx.ingress.kubernetes.io" denied the request: host "jimh-naccel-background.dev.ontariosystems.cloud" and path "/" is already defined in ingress jimh/jimh-naccel

Backup:  jimh-20210112182635

Namespaces:
  Included:  all namespaces found in the backup
  Excluded:  <none>

Resources:
  Included:        *
  Excluded:        nodes, events, events.events.k8s.io, backups.velero.io, restores.velero.io, resticrepositories.velero.io
  Cluster-scoped:  auto

Namespace mappings:  jimh=jimh2

Label selector:  <none>

Restore PVs:  auto

**volumesnapshots.json**
[{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-3b6d9e39-8d61-42ad-b68d-f9e9b4cf4b98","providerVolumeID":"vol-003d864be71562681","volumeType":"gp2"},"status":{"providerSnapshotID":"snap-0eb3246c0644cbe9d","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-09fb8590-dc17-4306-a6d7-b6f9a92cf865","providerVolumeID":"vol-06414a358b55ec702","volumeType":"gp2"},"status":{"providerSnapshotID":"snap-056e2854dc37a2b0e","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-1c34bcd8-12df-4e33-b603-c3d91348ee3d","providerVolumeID":"vol-02408e850f8aac83a","volumeType":"gp2","volumeAZ":"us-east-1c"},"status":{"providerSnapshotID":"snap-04cc450107494d26b","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-73763eac-c5d6-4e97-87cf-e04c2d915189","providerVolumeID":"vol-01ece1245890d277f","volumeType":"gp2"},"status":{"providerSnapshotID":"snap-086608e971df82eec","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-da3373d3-ef09-4d71-ab19-772dc25d7602","providerVolumeID":"vol-0709c1adfecabfc63","volumeType":"gp2","volumeAZ":"us-east-1c"},"status":{"providerSnapshotID":"snap-077b922cb6eb0c25d","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-524907f2-f8b5-4ed1-bd32-7ba56ea75cb6","providerVolumeID":"vol-0e45281f1fb242ad1","volumeType":"gp2"},"status":{"providerSnapshotID":"snap-0e6e039d37a0c6a8a","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-adc8fc17-6644-4b1a-b055-14009e05a179","providerVolumeID":"vol-0ca64ce0ea55e1610","volumeType":"gp2"},"status":{"providerSnapshotID":"snap-067504e6072e4d53a","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-f65272d1-b6c0-4579-b999-9f5622e3ede3","providerVolumeID":"vol-0edf1a1906b41e747","volumeType":"gp2"},"status":{"providerSnapshotID":"snap-0bdbc6bdb44819cdc","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-1d53494c-af48-4f01-8903-cc51f8781493","providerVolumeID":"vol-0793639a987a087cf","volumeType":"gp2","volumeAZ":"us-east-1b"},"status":{"providerSnapshotID":"snap-0e4800c1caef31cad","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-6c8d62ac-22dc-4d8f-af88-63ae3a492910","providerVolumeID":"vol-0fbb16b2ff01e3b7b","volumeType":"gp2","volumeAZ":"us-east-1b"},"status":{"providerSnapshotID":"snap-0e24c15a0597cea58","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-afc80ace-304a-4a1e-ae18-6dfbc2ccf777","providerVolumeID":"vol-083a4d892e3165744","volumeType":"gp2","volumeAZ":"us-east-1b"},"status":{"providerSnapshotID":"snap-00e0adeda6e7b5d1a","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-d406af94-b65f-40a5-b42c-3c84b60c6f5d","providerVolumeID":"vol-072810f50ec172407","volumeType":"gp2","volumeAZ":"us-east-1b"},"status":{"providerSnapshotID":"snap-000bf24342847268e","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-725c46fe-1da3-4b5c-9e37-f25cbf22f6f3","providerVolumeID":"vol-0c73d2e90faf4c6e6","volumeType":"gp2","volumeAZ":"us-east-1b"},"status":{"providerSnapshotID":"snap-02c56af601dc2e848","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-1d8a3ec0-57e7-4f36-af55-aaa9fe556fd4","providerVolumeID":"vol-0587f7ea3e4d97e21","volumeType":"gp2"},"status":{"providerSnapshotID":"snap-0ecf0bc5d1facb765","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-63ae7616-07b8-482e-9b78-b140424a2601","providerVolumeID":"vol-001c906b5fe2bab32","volumeType":"gp2"},"status":{"providerSnapshotID":"snap-080f2f9233de95c6d","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-1f1d1f49-4ff2-467c-868f-0b66f638811e","providerVolumeID":"vol-00a1e4c56302a3455","volumeType":"gp2","volumeAZ":"us-east-1c"},"status":{"providerSnapshotID":"snap-07fd903c779fdf88d","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-9eed36cf-054a-4c9c-9161-e68677e477dc","providerVolumeID":"vol-0d8e5dc396d802ddc","volumeType":"gp2","volumeAZ":"us-east-1c"},"status":{"providerSnapshotID":"snap-0943d31635df82082","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-b3016087-c4c6-49bf-aa35-2b388830e3dc","providerVolumeID":"vol-0ecb53b49c2a6d2b2","volumeType":"gp2","volumeAZ":"us-east-1c"},"status":{"providerSnapshotID":"snap-04ce68a76ab98211d","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-e7109c97-795d-4e92-92dc-2e012026c21a","providerVolumeID":"vol-0b5ac7721840883fe","volumeType":"gp2","volumeAZ":"us-east-1c"},"status":{"providerSnapshotID":"snap-07a85ead071d56ba7","phase":"Completed"}},{"spec":{"backupName":"jimh-20210112182635","backupUID":"f533a56b-8528-42da-8ac4-9c89050af3f1","location":"default","PersistentVolumeName":"pvc-6593de51-a66a-445d-a01c-5792aeff4f7e","providerVolumeID":"vol-036e7e3a5309361bc","volumeType":"gp2"},"status":{"providerSnapshotID":"snap-0fed0c13863eacf21","phase":"Completed"}}]

**Environment:**

- Velero version (use `velero version`): 
Client:
	Version: v1.5.2
	Git commit: e115e5a191b1fdb5d379b62a35916115e77124a4
Server:
	Version: v1.5.2

- Velero features (use `velero client config get features`): 
features: <NOT SET>

- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.14", GitCommit:"89182bdd065fbcaffefec691908a739d161efc03", GitTreeState:"clean", BuildDate:"2020-12-18T12:11:25Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.9-eks-d1db3c", GitCommit:"d1db3c46e55f95d6a7d3e5578689371318f95ff9", GitTreeState:"clean", BuildDate:"2020-10-20T22:18:07Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

- Cloud provider or hardware configuration:
AWS EKS running kube 1.18 (I believe 1.15 when these volumes were created)
 
- OS (e.g. from `/etc/os-release`):
Nodes are running Amazon Linux 2 (linux/amd64)

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by dsu-igeek** [comment]
The zone is going to be retrieved from either the `topology.kubernetes.io/zone` label or the `failure-domain.beta.kubernetes.io/zone` label (pre 1.17).  Would you check your PVs to see if this is set?  This is a PV in my test AWS cluster, we can see the label `failure-domain.beta.kubernetes.io/zone` is set here to us-west-2a

`kubectl get -o yaml `<PV name>

`apiVersion: v1
kind: PersistentVolume
metadata:
  annotations:
    kubernetes.io/createdby: aws-ebs-dynamic-provisioner
    pv.kubernetes.io/bound-by-controller: "yes"
    pv.kubernetes.io/provisioned-by: kubernetes.io/aws-ebs
  creationTimestamp: "2021-02-17T23:19:27Z"
  finalizers:
  - kubernetes.io/pv-protection
  labels:
    failure-domain.beta.kubernetes.io/region: us-west-2
    failure-domain.beta.kubernetes.io/zone: us-west-2a
  managedFields:
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:kubernetes.io/createdby: {}
          f:pv.kubernetes.io/bound-by-controller: {}
          f:pv.kubernetes.io/provisioned-by: {}
        f:labels:
          .: {}
          f:failure-domain.beta.kubernetes.io/region: {}
          f:failure-domain.beta.kubernetes.io/zone: {}
      f:spec:
        f:accessModes: {}
        f:awsElasticBlockStore:
          .: {}
          f:fsType: {}
          f:volumeID: {}
        f:capacity:
          .: {}
          f:storage: {}
        f:claimRef:
          .: {}
          f:apiVersion: {}
          f:kind: {}
          f:name: {}
          f:namespace: {}
          f:resourceVersion: {}
          f:uid: {}
        f:nodeAffinity:
          .: {}
          f:required:
            .: {}
            f:nodeSelectorTerms: {}
        f:persistentVolumeReclaimPolicy: {}
        f:storageClassName: {}
        f:volumeMode: {}
      f:status:
        f:phase: {}
    manager: kube-controller-manager
    operation: Update
    time: "2021-02-17T23:19:27Z"
  name: pvc-4b05135b-2a66-49d8-bb82-4985f7ebb7df
  resourceVersion: "22778664"
  selfLink: /api/v1/persistentvolumes/pvc-4b05135b-2a66-49d8-bb82-4985f7ebb7df
  uid: ff267b82-5c0a-492a-9234-7176a70f5d7c
spec:
  accessModes:
  - ReadWriteOnce
  awsElasticBlockStore:
    fsType: ext4
    volumeID: aws://us-west-2a/vol-053c0e388cbb77bd7
  capacity:
    storage: 1Gi
  claimRef:
    apiVersion: v1
    kind: PersistentVolumeClaim
    name: kibishii-data-kibishii-deployment-0
    namespace: kibishii-workload
    resourceVersion: "22778612"
    uid: 4b05135b-2a66-49d8-bb82-4985f7ebb7df
  nodeAffinity:
    required:
      nodeSelectorTerms:
      - matchExpressions:
        - key: failure-domain.beta.kubernetes.io/zone
          operator: In
          values:
          - us-west-2a
        - key: failure-domain.beta.kubernetes.io/region
          operator: In
          values:
          - us-west-2
  persistentVolumeReclaimPolicy: Delete
  storageClassName: kibishii-storage-class
  volumeMode: Filesystem
status:
  phase: Bound
`

**Reply by i300543** [reply]
we have the same issue, and the labels do not exist on the pv:
```
apiVersion: v1
kind: PersistentVolume
metadata:
  annotations:
    pv.kubernetes.io/provisioned-by: ebs.csi.aws.com
  creationTimestamp: "2021-05-20T05:29:48Z"
  finalizers:
  - kubernetes.io/pv-protection
  - external-attacher/ebs-csi-aws-com
  managedFields:
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:pv.kubernetes.io/provisioned-by: {}
      f:spec:
        f:accessModes: {}
        f:awsElasticBlockStore:
          .: {}
          f:fsType: {}
          f:volumeID: {}
        f:capacity:
          .: {}
          f:storage: {}
        f:claimRef:
          .: {}
          f:apiVersion: {}
          f:kind: {}
          f:name: {}
          f:namespace: {}
          f:resourceVersion: {}
          f:uid: {}
        f:nodeAffinity:
          .: {}
          f:required:
            .: {}
            f:nodeSelectorTerms: {}
        f:persistentVolumeReclaimPolicy: {}
        f:storageClassName: {}
        f:volumeMode: {}
    manager: csi-provisioner
    operation: Update
    time: "2021-05-20T05:29:48Z"
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:status:
        f:phase: {}
    manager: kube-controller-manager
    operation: Update
    time: "2021-05-20T05:29:48Z"
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:finalizers:
          v:"external-attacher/ebs-csi-aws-com": {}
    manager: csi-attacher
    operation: Update
    time: "2021-05-20T05:29:49Z"
  name: pv-shoot--devx-plat1--kojan2-0c29c5bf-f754-4681-8db5-c154cc6e5c64
  resourceVersion: "195655"
  selfLink: /api/v1/persistentvolumes/pv-shoot--devx-plat1--kojan2-0c29c5bf-f754-4681-8db5-c154cc6e5c64
  uid: 1c8f475b-4e1a-4a1b-80f0-127f6339de9a
spec:
  accessModes:
  - ReadWriteOnce
  awsElasticBlockStore:
    fsType: ext4
    volumeID: vol-064751c802e5faf02
  capacity:
    storage: 20Gi
  claimRef:
    apiVersion: v1
    kind: PersistentVolumeClaim
    name: redis-data-malws-redis-master-0
    namespace: malware-scan-agent
    resourceVersion: "195569"
    uid: 0c29c5bf-f754-4681-8db5-c154cc6e5c64
  nodeAffinity:
    required:
      nodeSelectorTerms:
      - matchExpressions:
        - key: topology.ebs.csi.aws.com/zone
          operator: In
          values:
          - eu-central-1a
        - key: topology.kubernetes.io/zone
          operator: In
          values:
          - eu-central-1a
  persistentVolumeReclaimPolicy: Delete
  storageClassName: appstudio-default
  volumeMode: Filesystem
status:
  phase: Bound
```
any suggestions on how to approach resolving this issue ? we tried using k8s v1.18.18, v1.19.10

**Reply by kojanabzah** [reply]
Regarding Sergey PV yaml above , any one knows who is using the labels in velero ? PV yaml must contain zone & region in the labels for the restore to work  ?

**Comment by AlwaySummit** [comment]
I got the same issue here. Could anyone check this please?

**Comment by RomanGrekov** [comment]
Same issue

**Comment by chrislinan** [comment]
Hi all, here is the root cause: https://github.com/vmware-tanzu/velero/issues/4332
And the fix is already merged but not released yet.
I pathed my velero with the latest commit, and the problem solved.
